src/main/java/com/synopsys/integration/detect/DetectConfigurationFactoryJava.java

```diff
@@ -0,0 +1,466 @@
+        Boolean testConnections = detectConfiguration.getValueOrDefault(DetectProperties.Companion.getDETECT_TEST_CONNECTION());
+        return new ProductBootOptions(ignoreFailures, testConnections);
+    }
+
+    public ConnectionDetails createConnectionDetails() throws DetectUserFriendlyException, InvalidPropertyException {
+        Boolean alwaysTrust = PropertyConfigUtils.getFirstProvidedValueOrDefault(detectConfiguration, DetectProperties.Companion.getBLACKDUCK_TRUST_CERT(), DetectProperties.Companion.getBLACKDUCK_HUB_TRUST_CERT());
+        List<String> proxyIgnoredHosts = PropertyConfigUtils.getFirstProvidedValueOrDefault(detectConfiguration, DetectProperties.Companion.getBLACKDUCK_PROXY_IGNORED_HOSTS(), DetectProperties.Companion.getBLACKDUCK_HUB_PROXY_IGNORED_HOSTS());
+        List<Pattern> proxyPatterns = proxyIgnoredHosts.stream()
+        .map(it -> Pattern.compile(it))
+            .collect(Collectors.toList());
+        ProxyInfo proxyInformation = createBlackDuckProxyInfo();
+        return new ConnectionDetails(proxyInformation, proxyPatterns, findTimeoutInSeconds(), alwaysTrust);
+    }
+
+    public BlackDuckConnectionDetails createBlackDuckConnectionDetails() throws InvalidPropertyException, DetectUserFriendlyException {
+        Boolean offline = PropertyConfigUtils.getFirstProvidedValueOrDefault(detectConfiguration, DetectProperties.Companion.getBLACKDUCK_OFFLINE_MODE(), DetectProperties.Companion.getBLACKDUCK_HUB_OFFLINE_MODE());
+        String blackduckUrl = PropertyConfigUtils.getFirstProvidedValueOrEmpty(detectConfiguration, DetectProperties.Companion.getBLACKDUCK_URL(), DetectProperties.Companion.getBLACKDUCK_HUB_URL()).orElse(null);
+        Set<String> allBlackDuckKeys = new BlackDuckServerConfigBuilder().getPropertyKeys().stream()
+                                           .filter(it -> !(it .contains("proxy")))
+                                           .collect(Collectors.toSet());
+        Map<String, String> blackDuckProperties = detectConfiguration.getRaw(allBlackDuckKeys);
+
+        return new BlackDuckConnectionDetails(offline, blackduckUrl, blackDuckProperties, findParallelProcessors(), createConnectionDetails());
+    }
+    //#endregion
+
+    public PolarisServerConfigBuilder createPolarisServerConfigBuilder(File userHome) throws InvalidPropertyException {
+        PolarisServerConfigBuilder polarisServerConfigBuilder = PolarisServerConfig.newBuilder();
+        Set<String> allPolarisKeys = polarisServerConfigBuilder.getPropertyKeys();
+        Map<String, String> polarisProperties = detectConfiguration.getRaw(allPolarisKeys);
+
+        // Detect and polaris-common use different property keys for the Polaris URL,
+        // so we need to pull it from they Detect config using Detect's key,
+        // and write it to the polaris-common config using the polaris-common key.
+        Optional<String> polarisUrlValue = detectConfiguration.getRaw(DetectProperties.Companion.getPOLARIS_URL());
```

Comment:
Small but should do the orElse(null) on this line. String polarisUrlValue = ...orElse(null);